README.md
# cool_alert_two

A Flutter package to display animated alert dialogs.

NOTE: this is a forked version of the original cool_alert plugin which was no longer being actively maintained at the time of writing!

Cool Alert Two includes various bug fixes and improvements that I use in my own projects and which I am aiming to maintain as a public fork.
Feel free to contribute!

## Usage

To use this package, add cool_alert_two as a dependency in your pubspec.yaml file.
And add this import to your file.

import 'package:cool_alert_two/cool_alert_two.dart';

## Image
![Screenshot  Gif](screenshots/gif.gif)

### Example

```
CoolAlertTwo.show(
   context: context,
   type: CoolAlertType.success,
   text: "Your transaction was successful!",
);
```


### CoolAlertTwo Class

| Attribute        | Data type           | Description                                                                                                                                                   |            Default Value            |
|:----------------------|:-------------------|:--------------------------------------------------------------------------------------------------------------------------------------------------------------|:-----------------------------------:|
| context| BuildContext | @required                                                                                      |                Null                 |
| type | CoolAlertType | @required - Type of alert dialog, ex: CoolAlertType.success for success dialogs | Null
title| String | Set a custom title for dialog                                                                                                |                Based on the CoolAlertType selected|                 |
| text| String             | Set the description text of the dialog.                                                                                                                       |                Null                 |
| widget| Widget             | Set any you expect widget of the dialog.                                                                                                                       |                Null                 |
| confirmBtnText | String             | Text of confirm button                                                                                                                                       |                'Ok'                 |                 |
| confirmBtnTap| Function           | Function that handle click of confirm button                                                                       |                () => Navigator.pop(context)|
| confirmBtnColor| Color              | Color of confirm Button                                                                                                                                       |          Theme.of(context).primaryColor          |                |
| cancelBtnText| String             | Text of cancel button                                                                                                                                       |              'Cancel'               |
| cancelBtnTap| Function           | Function that handle click of cancel button                                                                     |                () => Navigator.pop(context) 
| barrierDismissible| bool               | Dismiss dialog on touch overlay                                                                                                                              |                true                 
| animType              | CoolAlertAnimType| Type of dialogue enter animation                                                                                                                              |           CoolAlertAnimType.scale|
| backgroundColor              | Color| Background color of the animation                                                                                                                              |           Color(0xFF515C6F)|
| confirmBtnTextStyle              | TextStyle | Confirm button text theme                                                                                                                              |           TextStyle(color: Colors.white, fontWeight:FontWeight.w600,fontSize: 18.0)|
| cancelBtnTextStyle              | TextStyle | Cancel button text theme                                                                                                                             |           TextStyle(color: Colors.grey, fontWeight:FontWeight.w600,fontSize: 18.0)|
| flareAsset              | String | Custom flare asset                                                                                                                             |           "animation.flr" |
| flareAnimationName              | String | The name of the flare animation to play                                                                                                                             |           "play" |
| lottieAsset              | String | Custom lottie asset                                                                                                                             |           "animation.json" |
|autoCloseDuration|Duration|Determines how long the dialog stays open for before closing|Null|
|width|double|Dialog width|MediaQuery.of(context).size.width|
|loopAnimation|boolean|Determines if the animation should loop or not|false|










CHANGELOG.md
## [0.1.1] - 26/01/2023

- Refactored package to consistently use the CoolAlertTwo namespace

### BREAKING CHANGE

All references to CoolAlert* classes should be refactored to CoolAlertTwo*

EG. new CoolAlert() >>> new CoolAlertTwo()

## [0.0.2] - 26/01/2023

- Updated Lottie to latest compatible version

## [0.0.1] - 17/01/2023

- Initial release
- Fork from cool_alert including bug fixes and improvements


lib/cool_alert.dart
library cool_alert;

import 'dart:async';

import 'package:flutter/material.dart';

import 'src/models/cool_alert_options.dart';
import 'src/utils/animate.dart';
import 'src/widgets/cool_alert_container.dart';

enum CoolAlertType { success, error, warning, confirm, info, loading, custom }

enum CoolAlertAnimType {
  scale,
  rotate,
  slideInDown,
  slideInUp,
  slideInLeft,
  slideInRight,
}

/// CoolAlert.
class CoolAlert {
  static Future show({
    /// BuildContext
    required BuildContext context,

    /// Title of the dialog
    String? title,

    /// Text of the dialog
    String? text,

    /// Custom Widget of the dialog
    Widget? widget,

    // Alert type [success, error, warning, confirm, info, loading, custom]
    required CoolAlertType type,

    // Animation type  [scale, rotate, slideInDown, slideInUp, slideInLeft, slideInRight]
    CoolAlertAnimType animType = CoolAlertAnimType.scale,

    /// Barrier Dissmisable
    bool barrierDismissible = true,

    // Triggered when confirm button is tapped
    VoidCallback? onConfirmBtnTap,

    /// Triggered when cancel button is tapped
    VoidCallback? onCancelBtnTap,

    /// Confirmation button text
    String confirmBtnText = 'Ok',

    /// Cancel button text
    String cancelBtnText = 'Cancel',

    /// Color for confirm button
    Color confirmBtnColor = const Color(0xFF3085D6),

    /// TextStyle for confirm button
    TextStyle? confirmBtnTextStyle,

    /// TextStyle for cancel button
    TextStyle? cancelBtnTextStyle,

    /// TextStyle for title
    TextStyle? titleTextStyle,

    /// TextStyle for main text
    TextStyle? textTextStyle,

    /// Determines if cancel button is shown or not
    bool showCancelBtn = false,

    /// Dialog Border Radius
    double borderRadius = 10.0,

    /// Header background color
    Color backgroundColor = const Color(0xFF515C6F),

    /// Flare asset path
    String? flareAsset,

    /// Flare animation name
    String flareAnimationName = 'play',

    /// Asset path of your lottie file
    String? lottieAsset,

    /// Width of the dialog
    double? width,

    /// Text alignment for title and body, default to center
    TextAlign? textAlignment,

    /// Determines how long the dialog stays open for before closing
    /// [default] is null
    /// When it is null, it won't autoclose
    Duration? autoCloseDuration,

    /// Detemines if the animation loops or not
    bool loopAnimation = false,
  }) {
    if (autoCloseDuration != null) {
      Future.delayed(autoCloseDuration, () {
        Navigator.of(context, rootNavigator: true).pop();
      });
    }

    final options = CoolAlertOptions(
        title: title,
        text: text,
        widget: widget,
        type: type,
        animType: animType,
        barrierDismissible: barrierDismissible,
        onConfirmBtnTap: onConfirmBtnTap,
        onCancelBtnTap: onCancelBtnTap,
        confirmBtnText: confirmBtnText,
        cancelBtnText: cancelBtnText,
        confirmBtnColor: confirmBtnColor,
        confirmBtnTextStyle: confirmBtnTextStyle,
        cancelBtnTextStyle: cancelBtnTextStyle,
        showCancelBtn: showCancelBtn,
        borderRadius: borderRadius,
        backgroundColor: backgroundColor,
        flareAsset: flareAsset,
        flareAnimationName: flareAnimationName,
        lottieAsset: lottieAsset,
        width: width,
        loopAnimation: loopAnimation,
        textTextStyle: textTextStyle,
        titleTextStyle: titleTextStyle,
        textAlignment: textAlignment);

    final child = AlertDialog(
      contentPadding: EdgeInsets.zero,
      shape: RoundedRectangleBorder(
        borderRadius: BorderRadius.circular(borderRadius),
      ),
      content: CoolAlertContainer(
        options: options,
      ),
    );

    return showGeneralDialog(
      barrierColor: Colors.black.withOpacity(0.5),
      transitionBuilder: (context, anim1, __, widget) {
        switch (animType) {
          case CoolAlertAnimType.scale:
            return Animate.scale(child: child, animation: anim1);

          case CoolAlertAnimType.rotate:
            return Animate.rotate(child: child, animation: anim1);

          case CoolAlertAnimType.slideInDown:
            return Animate.slideInDown(child: child, animation: anim1);

          case CoolAlertAnimType.slideInUp:
            return Animate.slideInUp(child: child, animation: anim1);

          case CoolAlertAnimType.slideInLeft:
            return Animate.slideInLeft(child: child, animation: anim1);

          case CoolAlertAnimType.slideInRight:
            return Animate.slideInRight(child: child, animation: anim1);

          default:
            return child;
        }
      },
      transitionDuration: Duration(milliseconds: 200),
      barrierDismissible:
          autoCloseDuration != null ? false : barrierDismissible,
      barrierLabel: '',
      context: context,
      pageBuilder: (context, _, __) => Container(),
    );
  }
}


lib/cool_alert_two.dart
library cool_alert_two;

import 'dart:async';

import 'package:flutter/material.dart';

import 'src/models/cool_alert_options.dart';
import 'src/utils/animate.dart';
import 'src/widgets/cool_alert_container.dart';

enum CoolAlertTwoType { success, error, warning, confirm, info, loading, custom }

enum CoolAlertTwoAnimType {
  scale,
  rotate,
  slideInDown,
  slideInUp,
  slideInLeft,
  slideInRight,
}

/// CoolAlertTwo.
class CoolAlertTwo {
  static Future show({
    /// BuildContext
    required BuildContext context,

    /// Title of the dialog
    String? title,

    /// Text of the dialog
    String? text,

    /// Custom Widget of the dialog
    Widget? widget,

    // Alert type [success, error, warning, confirm, info, loading, custom]
    required CoolAlertTwoType type,

    // Animation type  [scale, rotate, slideInDown, slideInUp, slideInLeft, slideInRight]
    CoolAlertTwoAnimType animType = CoolAlertTwoAnimType.scale,

    /// Barrier Dissmisable
    bool barrierDismissible = true,

    // Triggered when confirm button is tapped
    VoidCallback? onConfirmBtnTap,

    /// Triggered when cancel button is tapped
    VoidCallback? onCancelBtnTap,

    /// Confirmation button text
    String confirmBtnText = 'Ok',

    /// Cancel button text
    String cancelBtnText = 'Cancel',

    /// Color for confirm button
    Color confirmBtnColor = const Color(0xFF3085D6),

    /// TextStyle for confirm button
    TextStyle? confirmBtnTextStyle,

    /// TextStyle for cancel button
    TextStyle? cancelBtnTextStyle,

    /// TextStyle for title
    TextStyle? titleTextStyle,

    /// TextStyle for main text
    TextStyle? textTextStyle,

    /// Determines if cancel button is shown or not
    bool showCancelBtn = false,

    /// Dialog Border Radius
    double borderRadius = 10.0,

    /// Header background color
    Color backgroundColor = const Color(0xFF515C6F),

    /// Flare asset path
    String? flareAsset,

    /// Flare animation name
    String flareAnimationName = 'play',

    /// Asset path of your lottie file
    String? lottieAsset,

    /// Width of the dialog
    double? width,

    /// Text alignment for title and body, default to center
    TextAlign? textAlignment,

    /// Determines how long the dialog stays open for before closing
    /// [default] is null
    /// When it is null, it won't autoclose
    Duration? autoCloseDuration,

    /// Detemines if the animation loops or not
    bool loopAnimation = false,
  }) {
    if (autoCloseDuration != null) {
      Future.delayed(autoCloseDuration, () {
        Navigator.of(context, rootNavigator: true).pop();
      });
    }

    final options = CoolAlertTwoOptions(
        title: title,
        text: text,
        widget: widget,
        type: type,
        animType: animType,
        barrierDismissible: barrierDismissible,
        onConfirmBtnTap: onConfirmBtnTap,
        onCancelBtnTap: onCancelBtnTap,
        confirmBtnText: confirmBtnText,
        cancelBtnText: cancelBtnText,
        confirmBtnColor: confirmBtnColor,
        confirmBtnTextStyle: confirmBtnTextStyle,
        cancelBtnTextStyle: cancelBtnTextStyle,
        showCancelBtn: showCancelBtn,
        borderRadius: borderRadius,
        backgroundColor: backgroundColor,
        flareAsset: flareAsset,
        flareAnimationName: flareAnimationName,
        lottieAsset: lottieAsset,
        width: width,
        loopAnimation: loopAnimation,
        textTextStyle: textTextStyle,
        titleTextStyle: titleTextStyle,
        textAlignment: textAlignment);

    final child = AlertDialog(
      contentPadding: EdgeInsets.zero,
      shape: RoundedRectangleBorder(
        borderRadius: BorderRadius.circular(borderRadius),
      ),
      content: CoolAlertTwoContainer(
        options: options,
      ),
    );

    return showGeneralDialog(
      barrierColor: Colors.black.withOpacity(0.5),
      transitionBuilder: (context, anim1, __, widget) {
        switch (animType) {
          case CoolAlertTwoAnimType.scale:
            return Animate.scale(child: child, animation: anim1);

          case CoolAlertTwoAnimType.rotate:
            return Animate.rotate(child: child, animation: anim1);

          case CoolAlertTwoAnimType.slideInDown:
            return Animate.slideInDown(child: child, animation: anim1);

          case CoolAlertTwoAnimType.slideInUp:
            return Animate.slideInUp(child: child, animation: anim1);

          case CoolAlertTwoAnimType.slideInLeft:
            return Animate.slideInLeft(child: child, animation: anim1);

          case CoolAlertTwoAnimType.slideInRight:
            return Animate.slideInRight(child: child, animation: anim1);

          default:
            return child;
        }
      },
      transitionDuration: Duration(milliseconds: 200),
      barrierDismissible:
          autoCloseDuration != null ? false : barrierDismissible,
      barrierLabel: '',
      context: context,
      pageBuilder: (context, _, __) => Container(),
    );
  }
}


lib/src/models/cool_alert_options.dart
import 'package:cool_alert_two/cool_alert_two.dart';
import 'package:flutter/widgets.dart';

class CoolAlertTwoOptions {
  String? title;
  TextStyle? titleTextStyle;
  TextStyle? textTextStyle;
  TextAlign? textAlignment;
  String? text;
  Widget? widget;
  CoolAlertTwoType type;
  CoolAlertTwoAnimType? animType;
  bool? barrierDismissible = false;
  VoidCallback? onConfirmBtnTap;
  VoidCallback? onCancelBtnTap;
  String? confirmBtnText;
  String? cancelBtnText;
  Color? confirmBtnColor;
  TextStyle? confirmBtnTextStyle;
  TextStyle? cancelBtnTextStyle;
  bool? showCancelBtn;
  double? borderRadius;
  Color? backgroundColor;
  String? flareAsset;
  String? flareAnimationName;
  String? lottieAsset;
  double? width;
  bool loopAnimation;

  CoolAlertTwoOptions(
      {this.title,
      this.text,
      this.widget,
      required this.type,
      this.animType,
      this.barrierDismissible,
      this.onConfirmBtnTap,
      this.onCancelBtnTap,
      this.confirmBtnText,
      this.cancelBtnText,
      this.confirmBtnColor,
      this.confirmBtnTextStyle,
      this.cancelBtnTextStyle,
      this.showCancelBtn,
      this.borderRadius,
      this.backgroundColor,
      this.flareAsset,
      this.flareAnimationName,
      this.lottieAsset,
      this.width,
      this.loopAnimation = false,
      this.textTextStyle,
      this.titleTextStyle,
      this.textAlignment});
}


lib/src/widgets/cool_alert_buttons.dart
import 'package:cool_alert_two/cool_alert_two.dart';
import 'package:cool_alert_two/src/models/cool_alert_options.dart';
import 'package:flutter/material.dart';

class CoolAlertTwoButtons extends StatelessWidget {
  final CoolAlertTwoOptions? options;

  CoolAlertTwoButtons({
    Key? key,
    this.options,
  }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Container(
      margin: EdgeInsets.only(top: 10.0),
      child: Row(
        mainAxisAlignment: MainAxisAlignment.center,
        children: <Widget>[
          _cancelBtn(context),
          _okayBtn(context),
        ],
      ),
    );
  }

  Widget _okayBtn(context) {
    final showCancelBtn =
        options!.type == CoolAlertTwoType.confirm ? true : options!.showCancelBtn!;

    final _okayBtn = _buildButton(
      context: context,
      isOkayBtn: true,
      text: options!.confirmBtnText!,
      onTap: options!.onConfirmBtnTap ?? () => Navigator.pop(context),
    );

    if (showCancelBtn) {
      return Expanded(child: _okayBtn);
    } else {
      return _okayBtn;
    }
  }

  Widget _cancelBtn(context) {
    final showCancelBtn =
        options!.type == CoolAlertTwoType.confirm ? true : options!.showCancelBtn!;

    final _cancelBtn = _buildButton(
      context: context,
      isOkayBtn: false,
      text: options!.cancelBtnText!,
      onTap: options!.onCancelBtnTap ?? () => Navigator.pop(context),
    );

    if (showCancelBtn) {
      return Expanded(child: _cancelBtn);
    } else {
      return Container();
    }
  }

  Widget _buildButton({
    BuildContext? context,
    required bool isOkayBtn,
    required String text,
    VoidCallback? onTap,
  }) {
    final _btnText = Text(
      text,
      style: _defaultTextStyle(isOkayBtn),
    );

    final okayBtn = MaterialButton(
      shape: RoundedRectangleBorder(
        borderRadius: BorderRadius.circular(30.0),
      ),
      color: options!.confirmBtnColor ?? Theme.of(context!).primaryColor,
      onPressed: onTap,
      child: Container(
        child: Center(
          child: _btnText,
        ),
      ),
    );

    final cancelBtn = GestureDetector(
      onTap: onTap,
      child: Container(
        child: Center(
          child: _btnText,
        ),
      ),
    );

    return isOkayBtn ? okayBtn : cancelBtn;
  }

  TextStyle _defaultTextStyle(bool isOkayBtn) {
    final textStyle = TextStyle(
      color: isOkayBtn ? Colors.white : Colors.grey,
      fontWeight: FontWeight.w600,
      fontSize: 18.0,
    );

    if (isOkayBtn) {
      return options!.confirmBtnTextStyle ?? textStyle;
    } else {
      return options!.cancelBtnTextStyle ?? textStyle;
    }
  }
}


lib/src/widgets/cool_alert_container.dart
import 'package:cool_alert_two/cool_alert_two.dart';
import 'package:cool_alert_two/src/constants/images.dart';
import 'package:cool_alert_two/src/models/cool_alert_options.dart';
import 'package:cool_alert_two/src/utils/single_loop_controller.dart';
import 'package:cool_alert_two/src/widgets/cool_alert_buttons.dart';
import 'package:flare_flutter/flare_actor.dart';
import 'package:flutter/material.dart';
import 'package:lottie/lottie.dart';

class CoolAlertTwoContainer extends StatelessWidget {
  final CoolAlertTwoOptions? options;

  const CoolAlertTwoContainer({
    Key? key,
    this.options,
  }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    final _header = _buildHeader(context);
    final _title = _buildTitle(context);
    final _text = _buildText(context);
    final _buttons = _buildButtons();
    final _widget = _buildWidget(context);

    final _content = Container(
      padding: EdgeInsets.all(20.0),
      child: Column(
        children: [
          _title,
          SizedBox(
            height: 5.0,
          ),
          _text,
          _widget!,
          SizedBox(
            height: 10.0,
          ),
          _buttons
        ],
      ),
    );

    return Container(
      width: options!.width ?? MediaQuery.of(context).size.width,
      child: Column(
        mainAxisSize: MainAxisSize.min,
        children: [_header, _content],
      ),
    );
  }

  Widget _buildHeader(context) {
    if (options!.type == CoolAlertTwoType.loading) {
      return Container();
    } else {
      String? anim = AppAnim.success;

      switch (options!.type) {
        case CoolAlertTwoType.success:
          anim = AppAnim.success;
          break;
        case CoolAlertTwoType.error:
          anim = AppAnim.error;
          break;
        case CoolAlertTwoType.warning:
          anim = AppAnim.warning;
          break;
        case CoolAlertTwoType.confirm:
          anim = AppAnim.info;
          break;
        case CoolAlertTwoType.info:
          anim = AppAnim.info;
          break;
        default:
          anim = AppAnim.info;
      }

      if (options!.flareAsset != null) {
        anim = options!.flareAsset;
      }
      return Container(
        width: double.infinity,
        decoration: BoxDecoration(
          color: options!.backgroundColor,
          borderRadius: BorderRadius.only(
            topLeft: Radius.circular(options!.borderRadius!),
            topRight: Radius.circular(options!.borderRadius!),
          ),
        ),
        child: Container(
          height: 150,
          width: 150,
          child: options!.lottieAsset == null
              ? FlareActor(
                  anim,
                  animation: options!.loopAnimation
                      ? options!.flareAnimationName
                      : null,
                  controller: options!.loopAnimation
                      ? null
                      : SingleLoopController(
                          options!.flareAnimationName!,
                          1,
                        ),
                )
              : Lottie.asset(options!.lottieAsset!,
                  repeat: options!.loopAnimation),
        ),
      );
    }
  }

  Widget _buildTitle(context) {
    if (options!.type == CoolAlertTwoType.loading) {
      return Padding(
        padding: const EdgeInsets.only(bottom: 0.0),
        child: Container(
          height: 100,
          width: 100,
          child: options!.lottieAsset == null
              ? FlareActor(
                  AppAnim.loading,
                  animation: options!.flareAnimationName,
                )
              : Lottie.asset(options!.lottieAsset!),
        ),
      );
    } else {
      final title = options!.title ?? _whatTitle();
      return Visibility(
        visible: title != null,
        child: Text(
          '$title',
          textAlign: options!.textAlignment ?? TextAlign.center,
          style:
              options!.titleTextStyle ?? Theme.of(context).textTheme.titleLarge,
        ),
      );
    }
  }

  Widget _buildText(context) {
    if (options!.text == null && options!.type != CoolAlertTwoType.loading) {
      return Container();
    } else {
      String? text = '';
      if (options!.type == CoolAlertTwoType.loading) {
        text = options!.text ?? 'Loading...';
      } else {
        text = options!.text;
      }
      return Text(
        text ?? '',
        textAlign: options!.textAlignment ?? TextAlign.center,
        style: options!.textTextStyle,
      );
    }
  }

  Widget? _buildWidget(context) {
    if (options!.widget == null && options!.type != CoolAlertTwoType.custom) {
      return Container();
    } else {
      Widget widget = Container();
      if (options!.type == CoolAlertTwoType.custom) {
        widget = options!.widget ?? widget;
      }
      return options!.widget;
    }
  }

  Widget _buildButtons() {
    if (options!.type == CoolAlertTwoType.loading) {
      return Container();
    } else {
      return CoolAlertTwoButtons(
        options: options,
      );
    }
  }

  String? _whatTitle() {
    switch (options!.type) {
      case CoolAlertTwoType.success:
        return 'Success!!!';
      case CoolAlertTwoType.error:
        return 'Error!!!';
      case CoolAlertTwoType.warning:
        return 'Warning!!!';
      case CoolAlertTwoType.confirm:
        return 'Are you sure?';
      case CoolAlertTwoType.info:
        return 'Info!';
      case CoolAlertTwoType.custom:
        return null;
      case CoolAlertTwoType.loading:
        return null;
      default:
        return null;
    }
  }
}


example/lib/main.dart
import 'package:cool_alert_two/cool_alert_two.dart';
import 'package:flutter/material.dart';
import 'package:google_fonts/google_fonts.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Cool Alert',
      theme: ThemeData(
        primarySwatch: Colors.blue,
        fontFamily: GoogleFonts.poppins().fontFamily,
      ),
      home: MyHomePage(title: 'Cool Alert'),
      debugShowCheckedModeBanner: false,
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key? key, this.title}) : super(key: key);

  final String? title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    final successAlert = _buildButton(
      onTap: () {
        CoolAlertTwo.show(
          context: context,
          type: CoolAlertTwoType.success,
          text: 'Transaction completed successfully!',
          autoCloseDuration: Duration(seconds: 2),
        );
      },
      text: 'Success',
      color: Colors.green,
    );

    final errorAlert = _buildButton(
      onTap: () {
        CoolAlertTwo.show(
          context: context,
          type: CoolAlertTwoType.error,
          title: 'Oops...',
          text: 'Sorry, something went wrong',
          loopAnimation: false,
        );
      },
      text: 'Error',
      color: Colors.red,
    );

    final warningAlert = _buildButton(
      onTap: () {
        CoolAlertTwo.show(
          context: context,
          type: CoolAlertTwoType.warning,
          text: 'You just broke protocol',
        );
      },
      text: 'Warning',
      color: Colors.orange,
    );

    final infoAlert = _buildButton(
      onTap: () {
        CoolAlertTwo.show(
          context: context,
          type: CoolAlertTwoType.info,
          text: 'Buy two, get one free',
        );
      },
      text: 'Info',
      color: Colors.blue[100],
    );

    final confirmAlert = _buildButton(
      onTap: () {
        CoolAlertTwo.show(
          context: context,
          type: CoolAlertTwoType.confirm,
          text: 'Do you want to logout',
          confirmBtnText: 'Yes',
          cancelBtnText: 'No',
          confirmBtnColor: Colors.green,
        );
      },
      text: 'Confirm',
      color: Colors.lightGreen,
    );

    final loadingAlert = _buildButton(
      onTap: () {
        CoolAlertTwo.show(
          context: context,
          type: CoolAlertTwoType.loading,
        );
      },
      text: 'Loading',
      color: Colors.grey,
    );

    final customAlert = _buildButton(
      onTap: () {
        var _message = '';
        CoolAlertTwo.show(
          context: context,
          type: CoolAlertTwoType.custom,
          barrierDismissible: true,
          confirmBtnText: 'Save',
          widget: TextFormField(
            decoration: InputDecoration(
              hintText: 'Enter Phone Number',
              prefixIcon: Icon(
                Icons.phone_outlined,
              ),
            ),
            textInputAction: TextInputAction.next,
            keyboardType: TextInputType.phone,
            onChanged: (value) => _message = value,
          ),
          onConfirmBtnTap: () async {
            if (_message.length < 5) {
              await CoolAlertTwo.show(
                context: context,
                type: CoolAlertTwoType.error,
                text: 'Please input something',
              );
              return;
            }
            Navigator.pop(context);
            await Future.delayed(Duration(milliseconds: 1000));
            await CoolAlertTwo.show(
              context: context,
              type: CoolAlertTwoType.success,
              text: "Phone number '$_message' has been saved!.",
            );
          },
        );
      },
      text: 'Custom',
      color: Colors.orange,
    );

    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title!),
      ),
      body: SingleChildScrollView(
        padding: EdgeInsets.all(20.0),
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            successAlert,
            errorAlert,
            warningAlert,
            infoAlert,
            confirmAlert,
            loadingAlert,
            customAlert,
          ],
        ),
      ),
    );
  }

  Widget _buildButton({VoidCallback? onTap, required String text, Color? color}) {
    return Padding(
      padding: const EdgeInsets.only(bottom: 10.0),
      child: MaterialButton(
        color: color,
        minWidth: double.infinity,
        shape: RoundedRectangleBorder(
          borderRadius: BorderRadius.circular(30.0),
        ),
        onPressed: onTap,
        child: Padding(
          padding: const EdgeInsets.symmetric(vertical: 15.0),
          child: Text(
            text,
            style: TextStyle(
              color: Colors.white,
            ),
          ),
        ),
      ),
    );
  }
}
